### PR TITLE
Add test case to 'Wherefore art thou'

### LINKED ISSUE #14057: Wherefore art thou could use more test cases.
Challenge [Wherefore art thou](https://www.freecodecamp.com/challenges/wherefore-art-thou#?solution=%2F%2F%20jshint%20esversion%3A6%0A%0Afunction%20whatIsInAName(collection%2C%20source)%20%7B%0A%20%20%2F%2F%20What's%20in%20a%20name%3F%0A%20%20var%20arr%20%3D%20%5B%5D%3B%0A%20%20%2F%2F%20Only%20change%20code%20below%20this%20line%0A%20%20%0A%20%20%2F%2Fsearch%20through%20an%20array%20of%20objects%2C%20and%20return%20objects%20that%0A%20%20%2F%2Fcontain%20key%3Avalue%20pairs%20that%20are%20in%20the%20source%20object%0A%0A%20%20%0A%20%20collection.forEach(%0A%20%20%20%20(objectInCollection)%20%3D%3E%20%7Blet%20go%20%3D%20false%3B%20Object.keys(source).forEach(%0A%20%20%20%20%20%20(sourceObjectKey)%20%3D%3E%20%7B%20%0A%20%20%20%20%20%20%20%20if%20(objectInCollection.hasOwnProperty(sourceObjectKey)%20%26%26%20%0A%20%20%20%20%20%20%20%20%20%20%20%20source%5BsourceObjectKey%5D%20%3D%3D%20objectInCollection%5BsourceObjectKey%5D)%20%7B%0A%20%20%20%20%20%20%20%20%20%20go%20%3D%20true%3B%0A%20%20%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20else%0A%20%20%20%20%20%20%20%20%20%20go%20%3D%20false%3B%0A%20%20%20%20%20%20%7D)%3B%0A%20%20%20%20%20%20if%20(go%20%3D%3D%3D%20true)%0A%20%20%20%20%20%20%20%20arr.push(objectInCollection)%3B%0A%20%20%20%20%7D)%3B%0A%0A%20%20%2F%2F%20Only%20change%20code%20above%20this%20line%0A%20%20return%20arr%3B%0A%7D%0A%0Aconsole.log(whatIsInAName(%5B%7B%20first%3A%20%22Romeo%22%2C%20last%3A%20%22Montague%22%20%7D%2C%20%7B%20first%3A%20%22Mercutio%22%2C%20last%3A%20null%20%7D%2C%20%7B%20first%3A%20%22Tybalt%22%2C%20last%3A%20%22Capulet%22%7D%2C%20%7B%20%22a%22%3A%201%2C%20%22b%22%3A%202%2C%20%20%22c%22%3A%203%7D%5D%2C%20%7B%20%22a%22%3A%201%2C%20%22b%22%3A%209999%2C%20%20%22c%22%3A%203%7D))%3B%0A) seems lacking in test cases for invalid input.  I would think that 

```whatIsInAName([{ "a": 1, "b": 2,  "c": 3}], { "a": 1, "b": 9999,  "c": 3})```

ought to return an empty array, for instance.  There is currently no check for this, so my code which returns 

```{ "a": 1, "b": 2,  "c": 3}``` 

passes all test cases.

### PR DESCRIPTION
**Pull request by LukeBousfield**
## Pre-Submission Checklist

- [x] Branches much start with either `fix/` or `feature/`
- [x] You have only one commit (if not, [squash](https://github.com/FreeCodeCamp/         FreeCodeCamp/wiki/git-rebase#squashing-multiple-commits-into-one) them into one commit).
- [x] All new and existing tests pass `npm run test-challenges`.

## Description

Adds a test case to 'Wherefore art thou' ensuring incorrect solutions are not marked as correct.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Add new translation (feature adding new translations)

## Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue: Closes #14057

### CODE REVIEW

File: seed/challenges/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting.json [hunk 1]
```diff
@@ -277,7 +277,8 @@
...
+        "assert.deepEqual(whatIsInAName([{ \"apple\": 1, \"bat\": 2 }, { \"apple\": 1 }, { \"apple\": 1, \"bat\": 2, \"cookie\": 2 }, {\"bat\":2}], { \"apple\": 1, \"bat\": 2 }), [{ \"apple\": 1, \"bat\": 2 }, { \"apple\": 1, \"bat\": 2, \"cookie\":2 }], 'message: <code>whatIsInAName([{ \"apple\": 1, \"bat\": 2 }, { \"apple\": 1 }, { \"apple\": 1, \"bat\": 2, \"cookie\": 2 }, { \"bat\":2 }], { \"apple\": 1, \"bat\": 2 })</code> should return <code>[{ \"apple\": 1, \"bat\": 2 }, { \"apple\": 1, \"bat\": 2, \"cookie\":2 }]</code>.');",
+        "assert.deepEqual(whatIsInAName([{ \"a\": 1, \"b\": 2, \"c\": 3 }], { \"a\": 1, \"b\": 9999, \"c\": 3 }), [])"
```
Comment (erictleung) [inline seed/challenges/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting.json]: Can you add an appropriate message here for users to know what this test does? See the other tests in this challenge for examples.

Comment (CodingPower472) [inline seed/challenges/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting.json]: I will do that.

**Comment by CodingPower472** [comment]
Alright, I added the message.

**Comment by CodingPower472** [comment]
Sorry, there was an issue with the first commit.  I fixed the issue.

**Comment by CodingPower472** [comment]
What is the state of this pull request?

**Comment by CodingPower472** [comment]
@erictleung Done.

File: seed/challenges/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting.json [hunk 2]
```diff
@@ -277,7 +277,8 @@
...
+        "assert.deepEqual(whatIsInAName([{ \"apple\": 1, \"bat\": 2 }, { \"apple\": 1 }, { \"apple\": 1, \"bat\": 2, \"cookie\": 2 }, {\"bat\":2}], { \"apple\": 1, \"bat\": 2 }), [{ \"apple\": 1, \"bat\": 2 }, { \"apple\": 1, \"bat\": 2, \"cookie\":2 }], 'message: <code>whatIsInAName([{ \"apple\": 1, \"bat\": 2 }, { \"apple\": 1 }, { \"apple\": 1, \"bat\": 2, \"cookie\": 2 }, { \"bat\":2 }], { \"apple\": 1, \"bat\": 2 })</code> should return <code>[{ \"apple\": 1, \"bat\": 2 }, { \"apple\": 1, \"bat\": 2, \"cookie\":2 }]</code>.');",
+        "assert.deepEqual(whatIsInAName([{ \"a\": 1, \"b\": 2, \"c\": 3 }], { \"a\": 1, \"b\": 9999, \"c\": 3 }), [], 'message: <code>whatIsInAName([{\"a\": 1, \"b\": 2, \"c\": 3}], {\"a\": 1, \"b\": 9999, \"c\": 3})</code> should return <code>[]</code>')"
```
Comment (erictleung) [inline seed/challenges/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting.json]: Your ending is missing a semi-colon, which is being parsed wrong. The ending should be

```diff
- ...</code>')"
+ ...</code>.');"
```

**Review by erictleung — CHANGES_REQUESTED** [review]
Looks mostly good. Just make the change I've noted in the code and it should be good to go :+1:

File: seed/challenges/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting.json (hunk 2 above)
Comment (CodingPower472) [inline seed/challenges/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting.json]: When I look in my code, it seems this is already the case.

Comment (CodingPower472) [inline seed/challenges/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting.json]: Oops, never mind.

**Comment by CodingPower472** [comment]
Done.

**Review by erictleung — APPROVED** [review]
LGTM! :+1: Thanks for taking the time to contribute! :sparkles:

**Pull request merged**